### ISSUE TITLE
I have a question about the function "randomize_data_alltimes" in create_dataset.py

### ISSUE THREAD

**Issue by LeopoldACC** [Soumyabrata/CloudSegNet#2]
Thanks for your work on CloudSegNet and Swinyseg Dataset.My Issue is that how do you split the augmented dataset into train/val/test?
Because the augmented dataset contains 5 augmented samples for each sample in original dataset.
My question is that the shuffle operation is used to which one of 
```bash
1. all 6768 augmented data list
```
**Which may split one original img's augmented sample into train/val/test set at the same time.**
For example,0001.png in train set,[0001_1.png,0001_2.png,0001_3.png] in val set and [0001_4.png,0001_5.png] in test set. 
```bash
2. original 1128 data list,then read the relative aug data of them. 
```
**Which split one original img's augmented sample into just one of train/val/test**
For example, [0001.png,0001_1.png,0001_2.png,0001_3.png,0001_4.png,0001_5.png] will be in one of train/val/test set.  
?
Due to comparation should keep the setting of dataset with you,thanks if reply!

From my understanding of the code.
create_dayimages.py save the 6768 augmented data's image and GT into './data/day_images/day_scene_withAUG.h5'
which were get from
```python
# original scene image
h5f = h5py.File('./data/day_images/day_scene_withAUG.h5','r')
original_sceneimage = h5f['sceneimage'][:]
h5f.close()
print "original scene hdf5 file's shape", original_sceneimage.shape

# original ground truth image
h5f = h5py.File('./data/day_images/day_withAUG_GT.h5','r')
original_GTmasks = h5f['GTmasks'][:]
h5f.close()
print "original ground truth hdf5 file's shape", original_GTmasks.shape
original_GTmasks = original_GTmasks.astype('float32')/255.

(no_of_dayimages, _, _, _) = original_sceneimage.shape
(no_of_nightimages, _, _, _) = original_sceneimage_night.shape
```
And then concate the day and night data.
```python
# Combining both day and night images in a single tensor
scene_data = np.vstack([original_sceneimage,original_sceneimage_night])
print (scene_data.shape)

gt_data = np.vstack([original_GTmasks,original_GTmasks_night])
print (gt_data.shape)
``` 
So,the scene_data list contains the all 6768 augmented data list.
```python
(X_train, X_testing, Y_train, Y_testing, imagetype_testing) = randomize_data_alltimes(scene_data, gt_data, no_of_dayimages, no_of_nightimages, percentage_training=80, percentage_testing=20)
```

From my understanding of your process of creating dataset.Should I split the dataset as:shuffle all 6768 augmented data list and then split into train/val/test.

**Comment by Soumyabrata** [comment]
Thanks for the interest. It is advisable that if you use any particular image in your training dataset, I'd suggest not to use the variants of that particular training image in the testing set or validation set. All variants of that particular training image should be used in the same training set.
